actions/inventory.go

```diff
@@ -229,10 +229,11 @@ func (a *InventoryCollectorAction) Collect(ctx context.Context, device *common.D
 	// Update StorageControllerCollectors based on controller vendor attributes
 	if a.dynamicCollection {
 		for _, sc := range a.device.StorageControllers {
-			a.collectors.StorageControllerCollectors = append(
-				a.collectors.StorageControllerCollectors,
-				StorageControllerCollectorByVendor(sc.Vendor, a.trace),
-			)
+			c := StorageControllerCollectorByVendor(sc.Vendor, a.trace)
```

Comment:
This would be a bit easier to read in this form,

```suggestion
		if c := StorageControllerCollectorByVendor(sc.Vendor, a.trace); c != nil {
            a.collectors.StorageControllerCollectors = append(a.collectors.StorageControllerCollectors, c)
        }
```